File: src/NubipApp.Models/Domain/Curriculum.cs
using System.Collections.Generic;

namespace NubipApp.Models.Domain
{
    public class Curriculum
    {
        public int CurriculumId { get; set; }
        public int DepartamentId { get; set; }
        public int SubjectId { get; set; }
        public int LectionCount { get; set; }
        public int LabaratorCount { get; set; }
        public int Semester { get; set; }

        public Speciality Speciality { get; set; }
        public Departament Departament { get; set; }
        public Subject Subject { get; set; }
        
        public ICollection<Lesson> Lessons { get; set; }
    }
}

File: src/models/curriculum.ts
import mongoose from "mongoose";

let Schema = mongoose.Schema;

let CurriculumSchema = new Schema({
	curriculumId: Number,
	departamentId: [{
		type: Schema.Types.ObjectId, ref: 'Departament'
	}],
	subjectId: [{
		type: Schema.Types.ObjectId, ref: 'Subject'
	}],
	lectionCount: Number,
	labaratorCount: Number,
	semester: Number,
});

let Curriculum = mongoose.model('Curriculum', CurriculumSchema);

export default Curriculum;